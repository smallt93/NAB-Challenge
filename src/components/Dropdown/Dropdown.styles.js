import styled, { css } from "styled-components";

export const DropdownWrapper = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: ${({ theme }) => theme.rem(4)} ${({ theme }) => theme.rem(8)};
`;

export const DropdownItem = styled.li`
  padding: 0em .5em;
  line-height: 2rem;
  cursor: pointer;
  background-color: transparent;
  transition: all 200ms ease;
  border-radius: ${({ theme }) => theme.rem(3)};
  font-size: ${({ theme }) => theme.rem(10)};
  color: ${({ theme }) => theme.colors.textPrimary};
  ${({ theme, active }) =>
    active && css`
      background-color: ${theme.colors.greyHover};
    `
  }
`;