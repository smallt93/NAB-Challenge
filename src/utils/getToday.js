const weekDay = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];

export const getDayOfWeek = (date) => {
  const getDay = String(date.getDate()).padStart(2, '0');
  return `${weekDay[date.getDay()]} ${getDay}`;
};