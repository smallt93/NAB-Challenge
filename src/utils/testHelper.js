import React from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'stylesheets/defaultTheme.styles';


export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

export function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}
