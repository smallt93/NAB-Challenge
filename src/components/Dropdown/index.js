import React, { memo } from "react";
import { DropdownWrapper, DropdownItem } from './Dropdown.styles';

const Dropdown = ({
  data = [],
  activeId = 0,
  onSelectedItem = () => {},
  setActiveId = () => {},
}) => {
  const onFocusIn = (locationId) => {
    setActiveId(locationId)
  }

  const onFocusOut = () => {
    setActiveId(0)
  }

  return (
    <DropdownWrapper>
      {data.map((item, index) => (
        <DropdownItem
          key={index}
          onClick={() => onSelectedItem(item)}
          onMouseOver={() => onFocusIn(item.woeid)}
          onMouseOut={onFocusOut}
          active={activeId === item.woeid}
        >
          {item.title}
        </DropdownItem>
      ))}
    </DropdownWrapper>
  );
};

export default memo(Dropdown);
