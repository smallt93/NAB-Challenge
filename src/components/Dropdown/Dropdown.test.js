import React from 'react';
import { mountWithTheme as mount } from 'utils/testHelper';
import Dropdown from 'components/Dropdown';
import { DropdownItem } from 'components/Dropdown/Dropdown.styles';

describe('render Dropdown component', () => {
  let Component;

  const mockProps = {
    data: [{}, {}, {}],
    activeId: 0,
    onSelectedItem: jest.fn(),
    setActiveId: jest.fn(),
  }
  
  beforeEach(() => {
    Component = mount(<Dropdown {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    expect(Component.length).toBe(1);
  });

  it('should match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });

  it('should call setActiveId when mouse over and mouse out', () => {
    Component.setProps({
      data: [{}],
    });
    const DropdownItemComp = Component.find(DropdownItem);

    DropdownItemComp.simulate('mouseover');
    expect(mockProps.setActiveId).toHaveBeenCalledTimes(1);

    DropdownItemComp.simulate('mouseout');
    expect(mockProps.setActiveId).toHaveBeenCalledTimes(2);
    expect(mockProps.setActiveId).toHaveBeenCalledWith(0);
  });

  it('should call onSelectedItem when click dropdom item', () => {
    Component.setProps({
      data: [{}],
    });
    const DropdownItemComp = Component.find(DropdownItem);

    DropdownItemComp.simulate('click');
    expect(mockProps.onSelectedItem).toHaveBeenCalledTimes(1);
  });
});
