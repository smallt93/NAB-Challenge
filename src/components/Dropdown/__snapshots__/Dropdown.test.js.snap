// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Dropdown component should match snapshot 1`] = `ReactWrapper {}`;
